src/frontend/screens/JoinProject.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from "react";
+import { Alert, StyleSheet, View } from "react-native";
+import { FormattedMessage, defineMessages } from "react-intl";
+import QRCode from "react-native-qrcode-svg";
+import { getUniqueId } from "react-native-device-info";
+
+import useWifiStatus from "../hooks/useWifiStatus";
+import { MEDIUM_BLUE, WHITE } from "../lib/styles";
+import { BackIcon } from "../sharedComponents/icons";
+import Button from "../sharedComponents/Button";
+import HeaderTitle from "../sharedComponents/HeaderTitle";
+import IconButton from "../sharedComponents/IconButton";
+import Text from "../sharedComponents/Text";
+import WifiBar from "../sharedComponents/WifiBar";
+
+const m = defineMessages({
+  title: {
+    id: "screens.JoinProject.title",
+    defaultMessage: "Join a Project",
+  },
+  instructionsTitle: {
+    id: "screens.JoinProject.instructionsTitle",
+    defaultMessage: "Instructions",
+  },
+  instructionsDescription: {
+    id: "screens.JoinProject.instructionsDescription",
+    defaultMessage: "Show this QR code to your Project Admin",
+  },
+  sendJoinRequest: {
+    id: "screens.JoinProject.sendJoinRequest",
+    defaultMessage: "Send Join Request instead",
+  },
+});
+
+const JoinProject = () => {
+  const { ssid } = useWifiStatus();
+
+  const deviceName: string =
+    "Android " + getUniqueId().slice(0, 4).toUpperCase();
+
+  return (
+    <View style={styles.pageContainer}>
+      {/* TODO: Probably want to do something smarter here if no ssid is detected */}
+      {ssid && <WifiBar deviceName={deviceName} ssid={ssid} />}
```

Comment:
couple of questions here:

- do we want the wifi bar to be pressable here? e.g. to take them to system settings or something similar
- what should we show if no ssid is detected?

Reply:
ah i think [this mockup](https://www.figma.com/proto/ZOsM89L0xy0NUOCkuksJgq/OTF-%2B-UXFund?node-id=46062%3A501&scaling=scale-down-width&page-id=45808%3A2&hide-ui=1) probably answers the first question.

Reply:
> do we want the wifi bar to be pressable here? e.g. to take them to system settings or something similar

On the sync screen, pressing the wifi bar opens the wifi settings. I think the same behaviour here makes sense.

A slight improvement on this behaviour that I think would be clearer for the user (to avoid an unexpected jump to the phone settings app, e.g. when the user tries to press the back button in the top navigation bar and accidentally presses the wifi bar by mistake): Open a dialog/alert that explains why you would open the settings.

(A) No Wifi

Alert title: "No Wifi"
Alert message: "Open the settings app to turn on Wifi"
Alert OK: "Open Settings"
Alert cancel: "Cancel"

(B) Wifi network

Alert title: "Wifi: %network_name%"
Alert message: "You need to be on the same Wifi network to connect to other phones. Open the settings app to change your Wifi network"
Alert OK: "Open Settings"
Alert cancel: "Cancel"

Maybe Sabella has an idea here? I don't think this (additional alert dialog) is a blocker for merging this though.

Reply:
> what should we show if no ssid is detected?

Same as the sync modal I think, there is a "NoWifiBar" component.

Reply:
> Same as the sync modal I think, there is a "NoWifiBar" component.

Yeah I added support for that in [this PR](https://github.com/digidem/mapeo-mobile/blob/17bd81be6dd934914848188d6b15b80f93327e3f/src/frontend/sharedComponents/WifiBar.tsx) but will port to here